src/Simulator/IncrementalRedstoneSimulator/ObserverHandler.h

```diff
@@ -0,0 +1,128 @@
+
+#pragma once
+
+#include "RedstoneHandler.h"
+
+
+
+
+
+class cObserverHandler : public cRedstoneHandler
+{
+public:
+
+	inline static bool IsOn(NIBBLETYPE a_Meta)
+	{
+		return (a_Meta & 0x8) == 0x8;
+	}
+
+	inline static Vector3i Getಠ_ಠOffset(NIBBLETYPE a_Meta)
+	{
+		return -GetButtOffset(a_Meta);
+	}
+
+	inline static Vector3i GetButtOffset(NIBBLETYPE a_Meta)
+	{
+		switch (a_Meta & 0x7)
+		{
+			case 0x0: return { 0, 1, 0 };
+			case 0x1: return { 0, -1, 0 };
+			case 0x2: return { 0, 0, 1 };
+			case 0x3: return { 0, 0, -1 };
+			case 0x4: return { 1, 0, 0 };
+			case 0x5: return { -1, 0, 0 };
+			default:
+			{
+				LOGWARNING("%s: Unknown metadata: %d", __FUNCTION__, a_Meta);
+				ASSERT(!"Unknown metadata while determining orientation of observer!");
+				return { 0, 0, 0 };
+			}
+		}
+	}
+
+	static bool ShouldPowerOn(cWorld & a_World, const Vector3i & a_Position, NIBBLETYPE a_Meta, cIncrementalRedstoneSimulatorChunkData * a_Data)
+	{
+		BLOCKTYPE BlockType;
+		NIBBLETYPE BlockMeta;
+		if (!a_World.GetBlockTypeMeta(a_Position + Getಠ_ಠOffset(a_Meta), BlockType, BlockMeta))
+		{
+			return false;
+		}
+
+		auto ಠ_ಠ = PoweringData(BlockType, BlockMeta);
+		auto UpdateInfo = a_Data->ExchangeUpdateOncePowerData(a_Position, ಠ_ಠ);
+
+		// Determine if to signal an update based on the block previously observed changed
+		return (UpdateInfo.PoweringBlock != ಠ_ಠ.PoweringBlock) || (UpdateInfo.PowerLevel != ಠ_ಠ.PowerLevel);
```

Comment:
This was the line I was unsure about, does PoweringBlock change for every non-redstone block update observers react to?

Reply:
I took some... creative liberties to use PoweringBlock/PowerLevel to store BlockType(InFront)/BlockMeta. It's initially set on line 52.

Reply:
Such creative liberties need extra documentation. We don't want to end up with write-only code.